### PR TITLE
`pattern export` command for mesheryctl 

### LINKED ISSUE #10165: [Feat]: Support models export as OCI format
### Current Behavior

Presently Meshery users can not export models however we want to support exporting models in OCI (Open Container Initiative) format for enhancing sharing and portibility of models

This is child issue of #6447 

### Desired Behavior
Meshery clients should support exporting models in OCI format

### Acceptance Tests
- [ ] Meshery UI and mesheryctl should support of expoting models
- [ ] Update documentation for Meshery's support of exporting models as OCI format

### Mockups

---
### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### LINKED ISSUE #10195: Support OCI export of Designs, Patterns, Filters
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

### Current Behavior

Presently Meshery UI supports exporting as OCI (Open Container Initiative), Using mesheryctl, export in OCI format for enhancing sharing and portability of design.

This is child issue of #6447

### Desired Behavior

Mesheryctl should support exporting Designs, Patterns, Filters in OCI format

### Acceptance Tests

- [ ] mesheryctl should support the export of Designs, Patterns, Filters
- [ ] Update documentation

### Mockups

---

### Contributor Resources

- ⌨️ [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](http://discuss.meshery.io) and [Layer5 Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by Jougan-0**
**Notes for Reviewers**

This PR fixes #10195 
The pattern export command has a total of 3 flags.

1. --id :-  It fetches the pattern based on the id of the patter/design file.
2. --dtype :- Download type OCI and ahpkg(yaml).
3. --output :- Specify the output path where the fetched pattern/design file is stored.By default it is set to home/root directory.

The command takes into account the flags to be case insensitive that is --id and --ID is treated as same.
Documentation and index updated.

There is an additional feature to suggest the name if the sub-command is written wrong.
image

 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit 96a8c67958136232a52eb7a3639e28c44b2efb9f at: https://6681e231c6a6498f0357cb95--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit e00796b813f07c103e54a9bab87442ef23fe6ef4 at: https://6681e6b4c64fca68ee727bf1--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit c9d54d9fb19610a182113a5f899f0c0cb073adce at: https://6681e7f38770a6b1ffec8160--meshery-docs-preview.netlify.app

**Review by singh1203 — COMMENTED** [review]
Based on the issue, we need a subcommand for model export rather than pattern (Design) export.
Please let me know if I'm wrong.

**Comment by Jougan-0** [comment]
> Based on the issue, we need a subcommand for model export rather than pattern (Design) export. Please let me know if I'm wrong.

Yes, you are right I pushed the PR under the wrong issue.

**Comment by l5io** [comment]
🚀 Preview for commit e7ec83b2c48d3cf1a3f1b51b61d57da34744cda8 at: https://6682b511f3a7c706264c583f--meshery-docs-preview.netlify.app

**Review by leecalcote — DISMISSED** [review]
Good to go after merge conflict is resolved. 👍

**Comment by Deepak0320Singhal** [comment]
@Jougan-0 Thanks for your contribution.
Let's discuss this in Meshery Dev meeting on Wednesday. Adding this as an agenda item....
Meeting minutes: https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit?pli=1#
Meeting link: https://meet.layer5.io/meshery

**Comment by leecalcote** [comment]
@Jougan-0 merge conflict...

**Comment by l5io** [comment]
🚀 Preview for commit 6573a4e0de08431b4c1ff47a54d4c17b6173caff at: https://668484a7c8c86e357a16c967--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 307bb44ad86ecafc589871024137b95c3b6c1750 at: https://668487c39cd46537c5766c2e--meshery-docs-preview.netlify.app

File: docs/_data/mesheryctlcommands/cmds.yml
Comment (leecalcote) [inline docs/_data/mesheryctlcommands/cmds.yml:720]: Can you change this to `--type`?

**Review by leecalcote — CHANGES_REQUESTED** [review]

**Comment by leecalcote** [comment]
@Jougan-0 do we have the ability to export pattern by name?

**Comment by Jougan-0** [comment]
> @Jougan-0 do we have the ability to export pattern by name?

Nope, only by id.

File: docs/_data/mesheryctlcommands/cmds.yml
Comment (Jougan-0) [reply docs/_data/mesheryctlcommands/cmds.yml]: Done.

**Comment by l5io** [comment]
🚀 Preview for commit 04d84c154e55fed2da131d609a02c66e1fd3340f at: https://66848d41149d453d0112520d--meshery-docs-preview.netlify.app

File: mesheryctl/pkg/utils/helpers.go
Comment (leecalcote) [inline mesheryctl/pkg/utils/helpers.go:142]: @Jougan-0 I wonder if we can't delete all reference to "app" given it's prior deprecation.

Comment (Jougan-0) [reply mesheryctl/pkg/utils/helpers.go:142]: I can look into it. Opening a issue now.

Comment (Jougan-0) [reply mesheryctl/pkg/utils/helpers.go]: Done.

**Comment by l5io** [comment]
🚀 Preview for commit be430687d997dda94a7bbb543b04bfafe766df3f at: https://66849456756ef13f9b219f82--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 11648176e15af3ed9e3798c4600478e94c3a6ab1 at: https://6686b39dbc2b250ee8cc8418--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit acec6acf923327d07dbd3a35bafee31dd4f0f9ab at: https://6686b5c4ddec6c1febe883ac--meshery-docs-preview.netlify.app

**Comment by leecalcote** [comment]
> Reference in new

Thank you. To the extent that the offer a name that is not a precise match or one that is duplicative of another, are you providing a selection prompt for the user to select the specific pattern? If so, they'll need a couple additional pieces of information about the patterns in order to differentiate between the patterns and make a selection. We have examples of this sort of behavior in mesheryctl elsewhere. If this UX isn't included in this PR - that's fine - let's open a new issue for separate follow up by another contributor.

**Review by leecalcote — DISMISSED** [review]
Please confirm final UX when exporting by name. Open separate issue, if needed.

Ready for merge.

**Comment by Jougan-0** [comment]
> > Reference in new
> 
> Thank you. To the extent that the offer a name that is not a precise match or one that is duplicative of another, are you providing a selection prompt for the user to select the specific pattern? If so, they'll need a couple additional pieces of information about the patterns in order to differentiate between the patterns and make a selection. We have examples of this sort of behavior in mesheryctl elsewhere. If this UX isn't included in this PR - that's fine - let's open a new issue for separate follow up by another contributor.

Yes for same name we have selection prompt and for uniqueness we provide the name with the patternID.
image

**Comment by leecalcote** [comment]
Good work, @Jougan-0. I daresay that I'm impressed. Yes, please: created at, updated at, workspace (if associated), and owner are are quite distinguishing and helpful. 

Caution: Some of these details are not supported by all Providers.

**Comment by Jougan-0** [comment]
> Good work, @Jougan-0. I daresay that I'm impressed. Yes, please: created at, updated at, workspace (if associated), and owner are are quite distinguishing and helpful.
> 
> Caution: Some of these details are not supported by all Providers.
> 
> A tab-based or four whitespaces-based columnar layout for these details is ideal.

Roger that, captain. A tabular view with these information I will add them and update this PR. There are a number of issue that other contributors can chew up and this would need prior knowledge that would take time for other contributors. I'll update this PR with those columns quickly.

**Comment by Jougan-0** [comment]
@leecalcote couldn't quite figure out the workspace thing but here are the details we show.
image

**Comment by l5io** [comment]
🚀 Preview for commit bbbf0d9b8160f3e499dec6805592943ffc566fe8 at: https://6686cbdfdb1bcc3baeaff04f--meshery-docs-preview.netlify.app

**Pull request merged**
